core/cmd/local_client.go

```diff
@@ -527,7 +528,42 @@ func (cli *Client) PrepareTestDatabase(c *clipkg.Context) error {
 		return cli.errorOut(err)
 	}
 
-	return nil
+	return cli.errorOut(dropDanglingTestDBs(cli.Logger, db))
+}
+
+func dropDanglingTestDBs(lggr logger.Logger, db *sqlx.DB) (err error) {
+	// Drop all old dangling databases
+	var dbs []string
+	if err = db.Select(&dbs, `SELECT datname FROM pg_database WHERE datistemplate = false;`); err != nil {
+		return err
+	}
+
+	// dropping database is very slow in postgres so we parallelise it here
+	nWorkers := 25
+	ch := make(chan string)
+	wg := sync.WaitGroup{}
+	wg.Add(nWorkers)
+	errMu := sync.Mutex{}
+	for i := 0; i < nWorkers; i++ {
+		go func() {
+			defer wg.Done()
+			for dbname := range ch {
+				lggr.Infof("Dropping test database: %q", dbname)
+				gerr := utils.JustError(db.Exec(fmt.Sprintf(`DROP DATABASE IF EXISTS %s`, dbname)))
+				errMu.Lock()
+				err = multierr.Append(err, gerr)
+				errMu.Unlock()
```

Comment:
Maybe an irrelevant micro-optimization in this case, but for patterns like this where you don't abandon any work (nothing is cancelled), a buffered `chan error` (cap == len(dbs)) can be used in place of the `WaitGroup` and `Mutex`, then the main routine receives and `multierr.Append`s them.